File: bindings/csharp/Umicp.Core/Types/EncodingType.cs
using System.Text.Json.Serialization;

namespace Umicp.SDK.Types;

/// <summary>
/// Data encoding types for UMICP payloads
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum EncodingType
{
    /// <summary>
    /// 32-bit floating point encoding
    /// </summary>
    Float32,

    /// <summary>
    /// 64-bit floating point encoding
    /// </summary>
    Float64,

    /// <summary>
    /// 32-bit integer encoding
    /// </summary>
    Int32,

    /// <summary>
    /// 64-bit integer encoding
    /// </summary>
    Int64,

    /// <summary>
    /// UTF-8 text encoding
    /// </summary>
    Utf8,

    /// <summary>
    /// Base64 encoding
    /// </summary>
    Base64,

    /// <summary>
    /// Hexadecimal encoding
    /// </summary>
    Hex
}

File: bindings/kotlin/src/main/kotlin/com/hivellm/umicp/types/EncodingType.kt
package com.hivellm.umicp.types

import kotlinx.serialization.Serializable

/**
 * UMICP Encoding Types
 *
 * Defines the encoding format for payload data.
 */
@Serializable
enum class EncodingType {
    /** 32-bit floating point */
    FLOAT32,

    /** 64-bit floating point */
    FLOAT64,

    /** 32-bit signed integer */
    INT32,

    /** 64-bit signed integer */
    INT64,

    /** 8-bit unsigned integer */
    UINT8,

    /** 16-bit unsigned integer */
    UINT16,

    /** 32-bit unsigned integer */
    UINT32,

    /** 64-bit unsigned integer */
    UINT64;

    companion object {
        /**
         * Convert from string value
         */
        fun fromString(value: String): EncodingType? =
            entries.find { it.name.equals(value, ignoreCase = true) }
    }
}

